src/mlpack/tests/main_tests/nca_test.cpp

```diff
@@ -0,0 +1,380 @@
+/**
+ * @file nca_test.cpp
+ * @author Yasmine Dumouchel
+ *
+ * Test mlpackMain() of nca_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "nca";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/metrics/lmetric.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/nca/nca_main.cpp>
+#include <mlpack/core/optimizers/lbfgs/lbfgs.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct NCATestFixture
+{
+ public:
+  NCATestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~NCATestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(NCAMainTest, NCATestFixture);
+
+/**
+* Ensure that, when labels are implicitily given with input,
```

Comment:
Pedantic style issue; missing space at the beginning, this should align with the first ``*`` from the first line.